modules/reindex/src/test/java/org/elasticsearch/index/reindex/RestReindexActionTests.java

```diff
@@ -200,4 +209,41 @@ private RemoteInfo buildRemoteInfoHostTestCase(String hostInRest) throws IOExcep
 
         return RestReindexAction.buildRemoteInfo(source);
     }
+
+    /**
+     * test deprecation is logged if one or more types are used in source search request inside reindex
+     */
+    public void testTypeInSource() {
+        FakeRestRequest.Builder requestBuilder = new FakeRestRequest.Builder(xContentRegistry())
+                .withMethod(Method.POST)
+                .withPath("/_reindex");
+        final StringBuilder requestContent = new StringBuilder();
```

Comment:
Also just a personal preference, but for me it'd be easier to read if we used an `XContentBuilder` here to build up the source/ destination.

Reply:
I changed this to XContentBuilder.